taxi/forms.py

```diff
@@ -0,0 +1,52 @@
+from django import forms
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm
+from django.core.exceptions import ValidationError
+
+from taxi.models import Driver, Car
+
+
+class DriverForm(UserCreationForm):
+    class Meta(UserCreationForm):
+        model = Driver
+        fields = UserCreationForm.Meta.fields + (
+            "first_name",
+            "last_name",
+            "license_number",
+        )
+
+    def clean_license_number(self) -> str:
+        license_num = self.cleaned_data["license_number"]
+
+        if len(license_num) != 8:
+            raise ValidationError("License must consist of 8 characters")
+        if not license_num[:3].isupper() or not license_num[:3].isalpha():
+            raise ValidationError(
+                "The first three characters must be uppercase"
+            )
+        if not license_num[-5:].isdigit():
+            raise ValidationError(
+                "The last five characters must be digits"
+            )
+
+        return license_num
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    class Meta:
+        model = Driver
+        fields = ("license_number",)
+
+    def clean_license_number(self):
+        return DriverForm.clean_license_number(self)

```

Comment:
U can inherit from DriverForm to get this method here

Reply:
> U can inherit from DriverForm to get this method here

I tried in last commits, but tests failed.
![изображение](https://user-images.githubusercontent.com/123926866/228750761-b58fa082-341c-4ec2-a90b-f814f81dd50b.png)

As far as I understand, in such a case password is needed for licence number updating
![изображение](https://user-images.githubusercontent.com/123926866/228751359-69334af4-9b49-4aee-9c64-6c0aea2f5b60.png)


I am not sure, how it can be fixed.